rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -1119,40 +1124,49 @@ def multicall_specific(
             return output
         return [contract.decode(x, method_name, arguments[0]) for x in output]
 
-    def get_erc20_contract_info(self, address: ChecksumEvmAddress) -> dict[str, Any]:
-        """
-        Query an erc20 contract address and return basic information as:
-        - Decimals
-        - name
-        - symbol
-        At all times, the dictionary returned contains the keys; decimals, name & symbol.
-        Although the values might be None.
-        if it is provided in the contract. This method may raise:
-        - BadFunctionCallOutput: If there is an error calling a bad address
-        """
-        cache = self.contract_info_erc20_cache.get(address)
-        if cache is not None:
-            return cache
+    def get_multiple_erc20_contract_info(self, addresses: list[ChecksumEvmAddress]) -> dict[str, Any]:  # noqa: E501
+        all_info: dict[str, Any] = {}
 
-        properties = ('decimals', 'symbol', 'name')
-        info: dict[str, Any] = {}
+        if len(addresses) == 0:
+            return {}
 
-        abi = self.contracts.abi('ERC20_TOKEN')
-        contract = EvmContract(address=address, abi=abi, deployed_block=0)
-        try:
-            # Output contains call status and result
-            output = self.multicall_2(
-                require_success=False,
-                calls=[(address, contract.encode(method_name=prop)) for prop in properties],
-            )
-        except RemoteError:
-            # If something happens in the connection the output should have
-            # the same length as the tuple of properties
-            output = [(False, b'')] * len(properties)
+        contract = EvmContract(address=addresses[0], abi=self.contracts.abi('ERC20_TOKEN'), deployed_block=0)  # noqa: E501
+
+        for addresses_chunk in get_chunks(addresses, 8):  # chunk number seems to be highest that can work with etherscan url limit  # noqa: E501
+            calls = []
+            for address in addresses_chunk:
+                calls.extend([(address, contract.encode(method_name=prop)) for prop in ERC20_PROPERTIES])  # noqa: E501
```

Comment:
instead of extending multiple times isn't it better to do?

```suggestion
           calls = [
                [(address, contract.encode(method_name=prop)) for prop in ERC20_PROPERTIES])]
                for address in addresses_chunk
          ]
```

Reply:
yeah sure done. But as you also can see by the mistake you made (the code you wrote is invalid syntax) making multilist python list comprehensions is hard and error prone :wink: 

It's unnatural and hard to read. Heard they optimize it, so not bad to do it but still ... weird.